### DISCUSSION TITLE
Feed in Point

### DISCUSSION THREAD

**Discussion by Kuschla** [DarwinsBuddy/WienerNetzeSmartmeter#18] [General]
Hi, 

How can I share API Data of the feed-in point to support measurement of the power exported from a photovoltaic source?
would someone please guide me to be able to contribute? I'm not able to code, but can share data. 
 
thanks!

**Comment by DarwinsBuddy** [comment]
when logging into your smart meter web interface hit F12 in an arbitrary browser, select "Network" and navigate to a page showing the data you are interested in and which is suitable for presenting in homeassistant. export the requests+responses by right click on the individual ones (maybe filter first for Fetch/XHR, cause the others are not relevant for the rest).

Unfortunately, I do not have access to a photovoltaic source, and thus every implementation would be done in blindfold-mode. I can try. It would be easier with access to such a source for developing/testing. If you want to change your credentials, share them with me, and change'em back once I am done could also be a possiblity, though I understand if you don't want to trust strangers on the internet. :sweat_smile:

**Comment by mariomurrent-softwaresolutions** [comment]
@DarwinsBuddy So that in general means also the feed-in points are not available with your extension?

**Reply by DarwinsBuddy** [reply]
since this integration gained more attraction, people joined and helped decipher the different types of "Anlage type" of the API.
so pv could already be supported if someone with a PV would give us insight into the APIs responses. This whole project is reverse engineered (since there is no viable public API which we could use).

**Reply by mariomurrent-softwaresolutions** [reply]
Okay. Well if you need some data, let me know. You can add me on discord as well, maybe I can help with that: lightstalker89 is the username on discord

**Comment by DarwinsBuddy** [comment]
To make this more transparent, these are the different types of smartmeters we currently support.

[AnlagenTyp](https://github.com/DarwinsBuddy/WienerNetzeSmartmeter/blob/main/custom_components/wnsm/api/constants.py#L56)

If you encounter some smartmeters, that are not listed there or you don't see popping up in your home assistant, please Create a ticket with all the necessary information provided.
See [How can I add a new Smartmeter Type](https://github.com/DarwinsBuddy/WienerNetzeSmartmeter/wiki/FAQs#how-can-i-add-a-new-smartmeter-type) to contribute
